### ISSUE TITLE
Add ability to avoid shell processing

### ISSUE THREAD

**Issue by hwittenborn** [sharkdp/hyperfine#428]
I just got hyperfine installed on my system (sorry if I'm overlooking something obvious), but there doesn't seem to be any native way to avoid shell processing.

For example, I'd expect this to just print `${PWD}`, but alas it prints the currently directory of my terminal:

```sh
hyperfine 'echo ${PWD}' --show-output
```

This also leads to another gripe with hyperfine I'm having: I'm not wanting to put the entire command I'm wanting to use inside of quotes. I'd much prefer to do something like this, and have the arguments get passed to the command in question without processing by a shell:

```sh
hyperfine --show-output -- echo '${PWD}'
```

**Comment by sharkdp** [comment]
> but there doesn't seem to be any native way to avoid shell processing.

Please see #336 

> not wanting to put the entire command I'm wanting to use inside of quotes. I'd much prefer to do something like this, and have the arguments get passed to the command in question without processing by a shell:

It would be a possibility, but how would you do a multi-command benchmark? we would need some sort of separator argument:
```
hyperfine --show-output -- command1 --with --some --options \; command2 --with --other --options
```
which - to me - is even worse. Once we implement #336, we will only do simple string splitting for the arguments. So if you run `hyperfine 'echo ${PWD}'`, we will call `echo` with one argument `${PWD}` (literal).

**Comment by hwittenborn** [comment]
> we would need some sort of separator argument
...
which - to me - is even worse

The only place I'd want the above approach is when you **don't** use shell processing, as you could otherwise just pass the `;` via the command argument.

The linked #336 issue would honestly solve all of my needs though. I would just need things like `echo "hi me"` to still just be the single `hi me` argument passed to `echo`.

**Comment by sharkdp** [comment]
> The only place I'd want the above approach is when you **don't** use shell processing, as you could otherwise just pass the `;` via the command argument (inside the quotes).

Not sure if I explained it well. What I meant was: I want to compare the runtime of command1 with the runtime of command2. Right now, I can do it like this:
```bash
hyperfine --show-output 'command1 --with --some --options' 'command2 --with --other --options'
```

**Comment by sharkdp** [comment]
> The linked #336 issue would honestly solve all of my needs though. I would just need things like `echo "hi me"` to still just be the single `hi me` argument passed to `echo`.

Yes. We would probably use https://crates.io/crates/shell-words to do proper splitting of arguments.
